### PR TITLE
Synchronous keyword to actions

### LINKED ISSUE #890: [vocabulary] Consider adding keyword to describe synchronous actions
Related to #302 and also to https://github.com/w3c/wot-binding-templates/issues/2

**Use Case:** A Thing is used by a Consumer. Consumer sends requests sequentially after their response to invoke multiple actions. A more tangible version, Consumer sends an HTTP POST request to invoke an action, once the Thing replies with the response, the Consumer sends another one

**Problem:** Currently, it is not possible to describe whether an action is synchronous or asynchronous, i.e. whether the Thing responds once the action is finished or once it is accepted and started by the Thing. 

**Proposal:** I would suggest adding a new keyword for Action Affordance called something like `synchronous` or `sync` that has a boolean value. Since claiming that the action is synchronous is "more assertive", the default value of this keyword should be `false`, similar to the keywords `safe` and `idempotent`.

**Examples:** I have two devices that already exhibit this behavior:
- [TD of a pan tilt hat](https://wotify.org/library/Pan-Tilt%20HAT/thingdescription): This is a platform that can hold a camera and can be rotated. An action starts the rotation and the response is received once the movement is started.
- [TD of a robot arm](https://wotify.org/library/Dobot%20Magician%20with%20Sliding%20Rail/thingdescription): The robot arm receives an action request, goes and does the action (which can take a couple of seconds) and then responds. This one already has this keyword in its Action Affordances since we wanted to remember the behavior for ourselves.
Edit: see the illustration [in the comment below](https://github.com/w3c/wot-thing-description/issues/890#issuecomment-615486464).

It is important to note that the reason they are programmed like is due to the underlying library that is provided by the manufacturers.

**Alternatives:** I have looked at [SSN vocabulary](https://www.w3.org/TR/vocab-ssn/) but couldn't find a corresponding term.

**Note:** This will have some dependence on the protocol. For example, an MQTT based Thing cannot have synchronous actions unless it implements the response topic feature introduced in MQTT V5.0. However, we already have such protocol dependence sometimes, like readmultipleproperties which shouldn't be done with an HTTP GET request; observable properties and events with HTTP without subprotocol.

### PR DESCRIPTION
**Pull request by egekorkan**
fixes #890 

For some reason, I cannot get the keyword to show up in the action affordance table but its description shows up. Also, the SVG is rendered wrong

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1491.html" title="Last updated on May 18, 2022, 3:50 PM UTC (4add836)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1491/78cfc50...4add836.html" title="Last updated on May 18, 2022, 3:50 PM UTC (4add836)">Diff</a>

### CODE REVIEW

**Comment by thjaeckle** [comment]
`sync` is missing from the [Vocabulary term](https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1491/a1c662e...223876d.html#dfn-vocab-term) column in the table.

Great addition, that's also what Eclipse Ditto needs 👍

**Comment by egekorkan** [comment]
Thanks @thjaeckle . Do you mean that we should add a new terminology term?

Also, these type of actions are implemented in the HTTP Baseline profile as well and I forgot to mention it

**Comment by thjaeckle** [comment]
What I meant is that the column where the `sync` term should be is empty in the preview:
image

**Comment by egekorkan** [comment]
Ah yes, that is why I marked it as a draft. @sebastiankb could you help me with finding out how to introduce the `sync` keyword on the action affordance table?

**Comment by JKRhb** [comment]
Could it be that the keyword needs to also be included in `td-context.ttl`?

**Comment by egekorkan** [comment]
> Could it be that the keyword needs to also be included in `td-context.ttl`?

If I do that, npm run render deletes my changes so I am guessing that this file is auto generated?

**Comment by JKRhb** [comment]
> > Could it be that the keyword needs to also be included in `td-context.ttl`?
> 
> If I do that, npm run render deletes my changes so I am guessing that this file is auto generated?

Oh, sorry, you are right! I think it might actually be `td-context-1.1.jsonld`..?

**Comment by sebastiankb** [comment]
you need just to update the td-context.jsonld. I added this in the PR and should work now.

File: context/td-context.jsonld [hunk 1]
```diff
@@ -110,6 +110,9 @@
     "idempotent" : {
       "@id" : "td:isIdempotent"
     },
+    "isSync" : {
```
[the remark this answers is not in the document]
Comment (thjaeckle) [reply context/td-context.jsonld:113]: That's what I also would assume looking at the other terms `idempotent`, `safe` for actions ..

**Review by danielpeintner — COMMENTED** [review]
Question/Comment about naming.
Why do we use `isSync` while we do not use `isSafe` or `isIdempotent`?
I would argue to be consistent.

EDIT: I saw others mentioned this as well...

**Comment by sebastiankb** [comment]
short hint: if you change a semantic based definition, you typically need to touch 3 files:
- td-context.jsonld
- td-valildation.ttl
- td.ttl / hctl.ttl / wotsec.ttl / jsonschema.ttl

File: context/td-context.jsonld (hunk 1 above)
[the remark this answers is not in the document]
Comment (JKRhb) [reply context/td-context.jsonld:113]: @sebastiankb Hmm, looking at the new diff, I think the sync and isSync need to be swapped?

**Review by sebastiankb — APPROVED** [review]

**Review by JKRhb — APPROVED** [review]

**Review by danielpeintner — APPROVED** [review]

**Comment by sebastiankb** [comment]
@egekorkan can you check the conflicts

**Review by mlagally — CHANGES_REQUESTED** [review]
This PR contains changes to normative language that seem to be unrelated.

**Comment by sebastiankb** [comment]
from today's TD call:
- there was a discussion about the "sync" naming and why not "synchronous" (most preferred one in the group) or "blocking" or "async"
- @egekorkan will provide an update

**Comment by egekorkan** [comment]
@danielpeintner I have changed to synchronous and also fixed merge conflicts

**Pull request merged**